src/wallet/wallet.ts

```diff
@@ -103,170 +142,442 @@ class HathorWalletServiceWallet extends EventEmitter {
+        // No utxos available for the requested filter
+        utxos = retUtxos;
+      } else if (newOptions.authority) {
+        // Requests an authority utxo, then I return the count of requested authority utxos
+        utxos = retUtxos.slice(0, newOptions.count);
+      } else {
+        // We got an array of utxos, then we must check if there is enough amount to fill the totalAmount
+        // and slice the least possible utxos
+        const ret = transaction.selectUtxos(retUtxos, newOptions.totalAmount!);
+        changeAmount = ret.changeAmount;
+        utxos = ret.utxos;
+      }
+    } else {
+      throw new WalletRequestError('Error requesting utxo.');
+    }
+    return { utxos, changeAmount };
+  }
+
+  /**
+   * Calculate sign message for auth token
+   *
+   * @memberof HathorWalletServiceWallet
+   * @inner
+   */
+  getAuthSign(timestamp: number): string {
+    const xpriv = wallet.getXPrivKeyFromSeed(this.seed, {passphrase: this.passphrase, networkName: this.network.name});
+    const derivedPrivKey = wallet.deriveXpriv(xpriv, '0\'');
+    const address = derivedPrivKey.publicKey.toAddress(this.network.getNetwork()).toString();
+
+    const message = new bitcore.Message(String(timestamp).concat(this.walletId!).concat(address));
+
+    return message.sign(derivedPrivKey.privateKey);
+  }
+
+  /**
+   * Validate that the wallet auth token is valid
+   * If it's not valid, requests a new one and update
+   *
+   * @memberof HathorWalletServiceWallet
+   * @inner
+   */
+  async validateAndRenewAuthToken() {
+    const now = new Date();
+    const timestampNow = Math.floor(now.getTime() / 1000);
+
+    const validateJWTExpireDate = (token) => {
+      const base64Url = token.split('.')[1];
+      const base64 = base64Url.replace('-', '+').replace('_', '/');
+      const decodedData = JSON.parse(Buffer.from(base64, 'base64').toString('binary'));
+
+      // If the token will expire in the next 60 seconds (or has already expired)
+      const delta = 60;
+      if (timestampNow + delta > decodedData.exp) {
+        return false;
+      }
+
+      return true;
+    }
+
+    if (!this.authToken || !validateJWTExpireDate(this.authToken)) {
+      const sign = this.getAuthSign(timestampNow);
+      const response = await walletApi.createAuthToken(timestampNow, this.xpub!, sign);
+      if (response.status === 200 && response.data.success === true) {
+        this.authToken = response.data.token;
+      }
+    }
+  }
+
   /**
    * Send a transaction from an array of outputs and inputs
    *
-   * @memberof HathorWallet
+   * @memberof HathorWalletServiceWallet
    * @inner
    */
-  async sendManyOutputsTransaction(outputs, options = { inputs: [], changeAddress: null }) {
-    return await this.sendTxProposal(outputs, options);
+  async sendManyOutputsTransaction(outputs: OutputRequestObj[], options: { inputs?: InputRequestObj[], changeAddress?: string } = {}): Promise<TxProposalUpdateResponseData> {
+    this.checkWalletReady();
```

Comment:
Why do we need this call? It was added as first step in many methods of this file.

Reply:
The wallet service will return error in any request if the wallet is not ready (i.e. if it's still creating, or error). Then I just have this method to prevent useless lambda calls when the wallet is not ready.

Reply:
Ok. I'd just rename it to `failIfWalletNotReady()`. What do you think?

Reply:
Done here https://github.com/HathorNetwork/hathor-wallet-lib/pull/214/commits/e2f5bb90b67e0466d052f0398baf969b7cb12b2c